features/step_definitions/controller_node/background_steps.rb

```diff
@@ -15,5 +15,11 @@
 end
 
 Given(/^the controller node is in "([^"]*)" state$/) do |ready|
+  # before checking for "ready" state, make sure the controller is not
+  # in an "unknown" state
+  20.times do
```

Comment:
That is not how an infinite loop is done in Ruby.  For a start it's a strange mix of Python and Ruby which would cause a syntax error in both languages, but even when fixed it will cause a Hound warning:

https://github.com/bbatsov/ruby-style-guide#infinite-loop